src/pages/Task.tsx

```diff
@@ -275,14 +252,12 @@ const TaskCore: React.FC = () => {
                 >
                   <FilesTables />
                 </Tab>
-                {showBuildBaronTab ? (
-                  <Tab name="Build Baron" id="task-build-baron-tab">
-                    <BuildBaron
-                      data={buildBaronData}
-                      error={buildBaronError}
-                      taskId={id}
-                    />
-                  </Tab>
+                {failedTask && execution !== undefined ? (
```

Comment:
I don't think you need to save failedTask in state since it is derived from the most recent data from `GetTaskQuery`

Reply:
I did that since it is undefined at first, and I want it to re-render once it becomes defined. Is that not necessary?

Reply:
the component is expected to rerender when the value for `data` is updated so you can access fields from the `data` object directly. If you log `task?.status?.includes("failed") || task?.status?.includes("timed-out")` or render it as an html in your jsx, you should see that the component is rendering the latest values of `data` so you can avoid saving them state.